llvm/lib/Target/RISCV/MCTargetDesc/RISCVELFStreamer.cpp

```diff
@@ -118,6 +147,19 @@ void RISCVTargetELFStreamer::finish() {
   }
 
   MCA.setELFHeaderEFlags(EFlags);
+
+  unsigned GNUNoteFlags = 0;
+
+  // check ZICFILP or ZICFISS with features
+  // TODO should we check with codegen enable ex. -mllvm
```

Comment:
This seems like it should be resolved before merging?

Reply:
Yes, I think we shouldn't just check if there is CFI extension is on, 
we should also check if the codegen options are really enabled.
Ex. following code in RISCVFrameLowering

@yetingk  Do you think so?

```
if (!STI.hasForcedSWShadowStack() && STI.hasStdExtZicfiss()) {
  BuildMI(MBB, MI, DL, TII->get(RISCV::SSPUSH)).addReg(RAReg);
    return;
  }
```